src/app/templates/TokenListItem.tsx

```diff
@@ -303,16 +307,21 @@ interface DefaultListItemLayoutProps<T extends TempleChainKind> {
   index?: number;
   balance: BigNumber;
   onClick?: MouseEventHandler<TokenListItemElement>;
+  useIsVisible?: boolean;
 }
 
 const DefaultListItemLayoutHOC = <T extends TempleChainKind>(
   networkKind: T,
-  AssetIconWithNetwork: FC<{ chainId: ChainId<T>; assetSlug: string; className?: string }>
+  AssetIconWithNetwork: FC<{
+    chainId: ChainId<T>;
+    assetSlug: string;
+    className?: string;
+  }>
 ) =>
   forwardRef<TokenListItemElement, PropsWithChildren<DefaultListItemLayoutProps<T>>>(
-    ({ children, assetSlug, assetName, className, network, index, balance, onClick }, ref) => {
+    ({ children, assetSlug, assetName, className, network, index, balance, onClick, useIsVisible }, ref) => {
       const { chainId } = network;
-      const isVisible = useIsItemVisible(index);
+      const visible = useIsVisible ? useIsItemVisible(index) : true;
```

Comment:
As for me, it's better to do without a conditional hook call at all costs. For example, you may write here something like `const defaultVisible = useIsItemVisible(index);` but use `defaultVisible || !useIsVisible` here: https://github.com/madfish-solutions/templewallet-extension/pull/1318/files#diff-f1f6cb597698050322a01e71d99747fb4935d3517edbd577199788e7a3890cdaR336